client/src/game/ui/tools/DiceTool.vue

```diff
@@ -70,157 +112,584 @@ function addSelector(selector: (typeof DxConfig.selectorOptions)[number]): void
 }
 
 function addLiteral(literal: (typeof literalOptions)[number]): void {
-    const value = Number.parseInt(literal);
-    DxHelper.addLiteral(input, value);
-}
+    const literalAsOperator = literal as (typeof DxConfig.symbolOptions[number]);
 
-function addSymbol(symbol: (typeof DxConfig.symbolOptions)[number]): void {
-    input.value.push({ type: DxSegmentType.Operator, input: symbol });
+    if (DxConfig.symbolOptions.includes(literalAsOperator)) {
+        input.value.push({ type: DxSegmentType.Operator, input: literalAsOperator });
+    } else {
+        const value = Number.parseInt(literal);
+        DxHelper.addLiteral(input, value);
+    }
 }
 
 function updateFromString(event: Event): void {
     input.value = diceState.raw.systems!["2d"].parse((event.target as HTMLInputElement).value);
 }
 
+function populateInputFromHistoryRoll(roll: DeepReadonly<RollResult<Part>>): void {
+    const parts = roll.parts;
+
+    let content = "";
+
+    for (const part of parts) {
+        content += part.input ?? "";
+    }
+    input.value = diceState.raw.systems!["2d"].parse(content);
+}
+function populateInputFromHistoryIndex(index: number): void {
+    const historyItem = diceState.reactive.history[index];
+    if (historyItem === null || historyItem === undefined) return;
+    populateInputFromHistoryRoll(historyItem.roll);
+}
 async function roll(): Promise<void> {
     if (inputText.value.length === 0) return;
 
-    const results = await diceTool.roll(
-        inputText.value,
-        use3d.value === "yes",
-        shareResult.value.toLowerCase() as DiceRollResult["shareWith"],
-    );
     clear();
-    diceSystem.showResults(results);
+    awaitingRoll.value = true;
+
+    const getResult = async (): Promise<RollResult<Part>> => {
```

Comment:
Why did this need to become a function?

Reply:
It didn't. Just my lack of javascript/typescript experience showing